src/main/java/nextstep/qna/domain/Question.java

```diff
@@ -81,12 +88,41 @@ public boolean isDeleted() {
         return deleted;
     }
 
-    public List<Answer> getAnswers() {
+    public Answers getAnswers() {
         return answers;
     }
 
     @Override
     public String toString() {
         return "Question [id=" + getId() + ", title=" + title + ", contents=" + contents + ", writer=" + writer + "]";
     }
+
+    public DeleteHistories delete(NsUser loginUser) throws CannotDeleteException {
+        validateDelete(loginUser);
+        this.deleted = true;
+        deleteAnswers(loginUser);
+        return makeDeletehistories();
+    }
+
+    private DeleteHistories makeDeletehistories() {
+        DeleteHistories deleteHistories = new DeleteHistories(makeDeleteHistory());
+        deleteHistories.addAllDeleteHistories(answers.deleteAllHistories());
+        return deleteHistories;
+    }
+
+    private void validateDelete(NsUser loginUser) throws CannotDeleteException {
+        if (!Objects.equals(loginUser, writer)) {
```

Comment:
```suggestion
        if (!isOwner(loginUser)) {
```
이렇게 if 안 조건을 메서드를 분리하여 가독성있게 변경할 수 있을 것 같습니다.